### PR TITLE
Migrate Adafruit Grand Central to GPIO V2

### PR DESCRIPTION
**Pull request by rtsuk**
# Summary
Patterned after the Metro M4 Express, with the pad numbers extracted
from the schematic and the CircuitPython pins.c file for
the board.

# Checklist
  - [x] `CHANGELOG.md` for the BSP or HAL updated
  - [x] All new or modified code is well documented, especially public items
  - [x] No new warnings or clippy suggestions have been introduced (see CI or check locally)

### CODE REVIEW

File: boards/grand_central_m4/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # Unreleased
 
+- Update to GPIO V2.
```
Comment (TDHolmes) [inline boards/grand_central_m4/CHANGELOG.md:3]: nit: you could be more broad, since `atsamd-hal-0.15` required more than just GPIO v2

**Review by TDHolmes — CHANGES_REQUESTED** [review]
Looks good overall! just some tweaks

File: boards/grand_central_m4/CHANGELOG.md (hunk 1 above)
Comment (rtsuk) [reply boards/grand_central_m4/CHANGELOG.md:3]: Could you recommend some wording?

File: boards/grand_central_m4/examples/neopixel_blink.rs [hunk 2]
```diff
@@ -0,0 +1,59 @@
+#![no_std]
```
[the remark this answers is not in the document]
Comment (rtsuk) [reply boards/grand_central_m4/examples/neopixel_blink.rs:1]: I just brought over the ones from the metro m4 that could work without additional hardware, so I guess the benefit is whatever it is there.

File: boards/grand_central_m4/src/lib.rs [hunk 3]
```diff
@@ -1,15 +1,607 @@
...
 
 pub use atsamd_hal as hal;
-
-pub use hal::common::*;
+pub use hal::ehal;
 pub use hal::pac;
-pub use hal::samd51::*;
 
+#[cfg(feature = "rt")]
+extern crate cortex_m_rt;
```
[the remark this answers is not in the document]
Comment (rtsuk) [reply boards/grand_central_m4/src/lib.rs:8]: I'll see if I can remove it.

File: boards/grand_central_m4/examples/neopixel_blink.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (TDHolmes) [reply boards/grand_central_m4/examples/neopixel_blink.rs:1]: Let's delete it if you don't mind. Seems like duplicate example code with no benefit

File: boards/grand_central_m4/CHANGELOG.md (hunk 1 above)
Comment (TDHolmes) [reply boards/grand_central_m4/CHANGELOG.md:3]: How about something like

> - update to `atsamd-hal-0.15` (`v2` drivers of peripherals and removal of deprecated things)

**Review by TDHolmes — APPROVED** [review]
Thanks!

**Pull request merged**